public/app/features/trails/DataTrailsHistory.tsx

```diff
@@ -1,43 +1,66 @@
 import { css, cx } from '@emotion/css';
 import { useMemo } from 'react';
 
-import { GrafanaTheme2 } from '@grafana/data';
+import { getTimeZoneInfo, GrafanaTheme2, InternalTimeZones, TIME_FORMAT } from '@grafana/data';
+import { convertRawToRange } from '@grafana/data/src/datetime/rangeutil';
 import {
-  SceneObjectState,
-  SceneObjectBase,
+  getUrlSyncManager,
   SceneComponentProps,
-  SceneVariableValueChangedEvent,
+  SceneObjectBase,
+  SceneObjectState,
   SceneObjectStateChangedEvent,
+  SceneObjectUrlValue,
+  SceneObjectUrlValues,
   SceneTimeRange,
   sceneUtils,
+  SceneVariableValueChangedEvent,
 } from '@grafana/scenes';
-import { useStyles2, Tooltip, Stack } from '@grafana/ui';
+import { Stack, Tooltip, useStyles2 } from '@grafana/ui';
 
 import { DataTrail, DataTrailState, getTopSceneFor } from './DataTrail';
+import { SerializedTrailHistory } from './TrailStore/TrailStore';
 import { reportExploreMetrics } from './interactions';
 import { VAR_FILTERS } from './shared';
 import { getTrailFor, isSceneTimeRangeState } from './utils';
 
 export interface DataTrailsHistoryState extends SceneObjectState {
   currentStep: number;
   steps: DataTrailHistoryStep[];
+  filtersApplied: string[];
 }
 
 export function isDataTrailsHistoryState(state: SceneObjectState): state is DataTrailsHistoryState {
   return 'currentStep' in state && 'steps' in state;
 }
 
+export function isDataTrailHistoryFilter(filter?: SceneObjectUrlValue): filter is string[] {
```

Comment:
the `is` syntax is new to me - cool!

Reply:
Here is a nice detailed article about all the type guards. https://camchenry.com/blog/typescript-type-guards

Reply:
thanks for sharing - i bookmarked!